### PR TITLE
scripts: build all: Add tigerlake.

### PR DESCRIPTION
**Pull request by lgirdwood**
Add tigerlake target to build script.

Signed-off-by: Liam Girdwood <liam.r.girdwood@linux.intel.com>

### CODE REVIEW

**Comment by ranj063** [comment]
@lgirdwood the xtensa core information seems to be different than the one @jajanusz sent me for TGL?

**Comment by lgirdwood** [comment]
> @lgirdwood the xtensa core information seems to be different than the one @jajanusz sent me for TGL?

Can you paste the details you have, I took the partial version info from core-isa.h. I will update.

**Comment by ranj063** [comment]
> Can you paste the details you have, I took the partial version info from core-isa.h. I will update.

@lgirdwood to get the TGL firmware built I made the following changes in the script. Other than the xtensa core info, the 2 main changes were the use of a private key and the use openssl version 1.0.*.
Apparently openssl1.1 doesnt work.

index 7bcd0a7a..e3216d04 100755
--- a/scripts/xtensa-build-all.sh
+++ b/scripts/xtensa-build-all.sh
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2018 Intel Corporation. All rights reserved.
 
-SUPPORTED_PLATFORMS=(byt cht bdw hsw apl cnl sue icl skl kbl imx8)
+SUPPORTED_PLATFORMS=(byt cht bdw hsw apl cnl sue icl skl kbl imx8 tgl)
 BUILD_ROM=no
 BUILD_DEBUG=no
 BUILD_FORCE_UP=no
@@ -88,6 +88,7 @@ fi
 
 if [ "x$USE_PRIVATE_KEY" == "xyes" ]
 then
+       RIMAGE_PRIVATE_KEY="/home/ranjani/Downloads/private-tgl.pem"
        if [ -z ${RIMAGE_PRIVATE_KEY+x} ]
        then
                echo "Error: No variable specified for RIMAGE_PRIVATE_KEY"
@@ -225,6 +226,16 @@ do
                XTENSA_TOOLS_VERSION="RF-2016.4-linux"
                HAVE_ROM='yes'
        fi
+       if [ $j == "tgl" ]
+       then
+               PLATFORM="tigerlake"
+               ARCH="xtensa-smp"
+               XTENSA_CORE="cavs2x_LX6HiFi3_2017_8"
+
+               ROOT="$pwd/../xtensa-root/$HOST"
+               XTENSA_TOOLS_VERSION="RG-2017.8-linux"
+               HAVE_ROM='yes'
+       fi
        if [ $j == "imx8" ]
        then
                PLATFORM="imx8"
@@ -276,7 +287,9 @@ do
        cmake -DTOOLCHAIN=$TOOLCHAIN \
                -DROOT_DIR=$ROOT \
                -DCMAKE_VERBOSE_MAKEFILE=ON \
-               ${PRIVATE_KEY_OPTION} \
+               –DMEU_OPENSSL=/usr/local/ssl/bin/openssl \
+               -DMEU_PATH=/home/ranjani/meu \
+               -DMEU_PRIVATE_KEY=/home/ranjani/Downloads/private-tgl.pem \
                ..
 
        make ${PLATFORM}_defconfig

File: scripts/xtensa-build-all.sh [hunk 1]
```diff
@@ -225,6 +225,16 @@ do
 		XTENSA_TOOLS_VERSION="RF-2016.4-linux"
 		HAVE_ROM='yes'
 	fi
+	if [ $j == "tgl" ]
+	then
+		PLATFORM="tigerlake"
+		ARCH="xtensa-smp"
+		XTENSA_CORE="X6H3CNL_2016_4_linux"
```
Comment (jajanusz) [inline scripts/xtensa-build-all.sh]: As @ranj063 wrote core is `cavs2x_LX6HiFi3_2017_8` and then tools need to be from 2017.

**Comment by jajanusz** [comment]
You write about 2 different things ;) @ranj063 meant xtensa core name and @lgirdwood wrote about core-ish.h (I took it exactly as it is from xtensa core + added only ifdef for xcc)

**Comment by lgirdwood** [comment]
@jajanusz thanks, btw if you have version with corect details I will close this PR and merge yours ?

**Comment by lgirdwood** [comment]
@jajanusz - ok I wil close this and wait for you official version - I should have the GCC TGL upstreamed by that point too.

**Pull request closed without merging**